### ISSUE TITLE
Big integers cannot be displayed correctly

### ISSUE THREAD

**Issue by KronicDeth** [lumen/lumen#456]
For big integers using Rust's `i64::min_value` for the negative number and using `i64::max_value` for the positive number, the negative number loses its sign becoming positive while the positive number somehow becomes `-1`.

## `init.erl`

```erlang
-module(init).
-export([start/0]).
-import(erlang, [display/1]).

start() ->
  display(-9223372036854775808),
  display(9223372036854775807).
```

## Actual

```
9223372036854775808
-1
```

## Expected

```
-9223372036854775808
9223372036854775807
```

**Comment by bitwalker** [comment]
Looks like this is an artifact of the `erlang:-/1` representation for negated numbers; I'm canonicalizing those so that the constants are properly negated. I haven't looked at the actual implementation of `erlang:-/1` in liblumen_otp, but I wonder if there was an issue there; and though an error was returned, since the calls to BIFs are not currently handling error sentinel values, the code just kept executing and printed the weird results. I'm investigating further.

**Comment by bitwalker** [comment]
Fixed in 44164e85
